New package: SagbiHomotopy v1.0.0

JuliaRegistrator opened the pull request:
- Registering package: SagbiHomotopy
- Repository: https://github.com/Barbarabetti/SagbiHomotopy.jl
- Created by: @Barbarabetti
- Version: v1.0.0
- Commit: 3de99a8981f304acaa93a2bf6583bd897dc46ea8
- Reviewed by: @Barbarabetti
- Reference: https://github.com/Barbarabetti/SagbiHomotopy.jl/issues/4
<!-- bf0c69308befbd3ccf2cc956ac8a4671113b158fd87a6d4e741b0173dab051a48f7dc43e593ffff4357bc647a3a835981dbaa69b100a7e3378eae5e2e682728c84c75ebdad7e99dd3b4c2ca0cdf80950b0c327dc543681d14e067984d55a491d977aac19727c02d195ba75347c9aca62f92c909b1b3e4d5dfbc14b1ce91d8f6a430b257ffaee91839cf8b210aa843415c25d2cabad4a2a3d91f7de73957a16c00ec132a22fbae6a6a38ad7b7b27af8cd -->

goerz commented:
Thank you for submitting your package! Could you add a little bit of documentation before registering? [At the very least](https://discourse.julialang.org/t/when-does-it-make-sense-to-post-a-new-package-to-the-julia-repository/108701/4), that would be a description of the package's purpose and a small usage example in the README. An important part of packages in General is that any potential user can figure out what the package is about and how to get started with using it. That is really difficult when there is no documentation.

Also, are you sure you want to register a version number `v1.0.0`? In the Julia ecosystem, we try to encourage the proper use of [semantic versioning](https://semver.org/spec/v2.0.0-rc.1.html). As a general rule, versions `<v1.0` are for initial development, when the API of the package isn't nailed down yet, the package does not have a lot of dependents, and "backwards compatibility" is not a major concern. For an initial registration, that's usually the situation your package is in. Thus, an initial `v0.1` is usually the way to go. The bar for registering an initial `v0.1` is not very high. Basically, just

* Must have a name that is appropriate for the scope of the package
* Must have some non-trivial functionality (no "placeholder" packages)
* Must have a README that explains the purpose of the package and gives a basic usage example (or a link to documentation that is sufficient to figure out how to get started)

For a version `v1.0`, you're saying that the package already has a stable API. That has some implications for what one would expect from that package. Personally, I tend to apply the following guidelines:

* Must _define_ the stable API. That generally means _complete_ documentation.
* Should have tests with reasonable coverage (on the order of 80%). Otherwise, there's no way to know if the package actually implements its API correctly, and no basis for figuring out whether a pull request is "breaking" or not, going forward.
* Should somewhat reasonably fill the described scope of the package. No _major_ gaps in functionality.

Even if you think your package is fairly polished already, I always recommend registering with an initial `v0.1` anyway. That way, you have a chance to find out whether there are any major flaws in the package once people start to use it. Once it has a few dependents and you really should start to worry about maintaining backwards compatibility, you can always make a `v1.0` release. That doesn't mean you absolutely have to make significant changes to the `v0.1` version. Just make sure that your package is fully documented as "stable" at that point, has good test coverage, and all the other things people would expect from a "stable" package.

JuliaRegistrator committed:
New package: SagbiHomotopy v1.0.0

UUID: 79f7b27e-1e7f-49b0-a183-48c80ed555cb
Repo: https://github.com/Barbarabetti/SagbiHomotopy.jl.git
Tree: 6d1bce364283ce21c888be4fa5969a2d5713d876

Registrator tree SHA: 17aec322677d9b81cdd6b9b9236b09a3f1374c6a

Barbarabetti commented:
[noblock]

goerz commented:
Thank you for updating your README! That looks great, and is pretty much exactly what I'd ask for in an initial registration at v0.1.

I noticed that in

> A detailed documentation is available at the MathRepo page [MathRepo page SagbiHomotopy](https://mathrepo.mis.mpg.de/SagbiHomotopy/).

the link to the documentation is currently broken. If that link will indeed contain a full API documentation, you should be good to go. Otherwise, I would recommend changing the version number to <1.0 until you have something in place that details which functions make up the "stable" API. I'd still recommend [Documenter](https://documenter.juliadocs.org/stable/) for API documentations. I'd also recommend having testing in place (see the red CI-failing badge in your README). Ultimately, this is up to you, whether you want to follow [semantic versioning](https://semver.org), or how exactly you interpret the requirements of a "stable" 1.0 version.

If you do want to change the version number, just make that change in the repo, and then register it again the same way you did for the original registration. It will create a new registration PR, and I can close this one in favor of the new one. Otherwise, let me know that you wish to keep the initial version at v1.0.

> @JuliaRegistrator register()

See the [FAQ entry on "I commented @JuliaRegistrator register on a pull request in the General registry, but nothing happened."](https://github.com/JuliaRegistries/General?tab=readme-ov-file#i-commented-juliaregistrator-register-on-a-pull-request-in-the-general-registry-but-nothing-happened)

> [noblock]

That by itself also does nothing. To unblock the registration, [_all_ comments have to contain [noblock]](https://github.com/JuliaRegistries/General?tab=readme-ov-file#automerge-is-blocked-by-one-of-my-comments-how-do-i-unblock-it). Which means _I_ have to add [noblock] to [my comment](https://github.com/JuliaRegistries/General/pull/131298#issuecomment-2891147546), after either I'm happy with state of the repo, or because you tell me that you want to proceed with v1.0

Barbarabetti commented:
The link at the MathRepo page will contain the detailed documentation, it will be online in the next days, so I would leave it as it is and I want to proceed with v1.0

[noblock]
